src/diff.c

```diff
@@ -1922,6 +1929,344 @@ diff_clear(tabpage_T *tp)
     tp->tp_first_diff = NULL;
 }
 
+/*
+ *  return true if the options are set to use diff linematch
+ */
+    int
+diff_linematch(diff_T *dp)
+{
+    if (!(diff_flags & DIFF_LINEMATCH))
+    {
+	return 0;
+    }
+    // are there more than three diff buffers?
+    int tsize = 0;
+    for (int i = 0; i < DB_COUNT; i++)
+    {
+	if ( curtab->tp_diffbuf[i] != NULL )
+	{
+	    // for the rare case (bug?) that the count of a diff block is negative, do
+	    // not run the algorithm because this will try to allocate a negative
+	    // amount of space and crash
+	    if (dp->df_count[i] < 0)
+	    {
+	      return FALSE;
+	    }
+	    tsize += dp->df_count[i];
+	}
+    }
+    // avoid allocating a huge array because it will lag
+    return tsize <= linematch_lines;
+}
+
+    int
+get_max_diff_length(const diff_T *dp)
+{
+  int maxlength = 0;
```

Comment:
This function uses two spaces for indentation.  As the modelines are now added to these files, you can reformat all the new code using "=".